lib/zlib.js

```diff
@@ -287,40 +237,29 @@ function Zlib(opts, mode) {
       opts.writableObjectMode = false;
     }
   }
+
   Transform.call(this, opts);
+  this._hadError = false;
   this.bytesWritten = 0;
-  this._handle = new binding.Zlib(mode);
+  this._handle = handle;
+  handle[owner_symbol] = this;
   // Used by processCallback() and zlibOnError()
-  this._handle[owner_symbol] = this;
-  this._handle.onerror = zlibOnError;
-  this._hadError = false;
-  this._writeState = new Uint32Array(2);
-
-  if (!this._handle.init(windowBits,
-                         level,
-                         memLevel,
-                         strategy,
-                         this._writeState,
-                         processCallback,
-                         dictionary)) {
-    throw new ERR_ZLIB_INITIALIZATION_FAILED();
-  }
-
+  handle.onerror = zlibOnError;
   this._outBuffer = Buffer.allocUnsafe(chunkSize);
   this._outOffset = 0;
-  this._level = level;
-  this._strategy = strategy;
+
   this._chunkSize = chunkSize;
   this._defaultFlushFlag = flush;
   this._finishFlushFlag = finishFlush;
   this._nextFlush = -1;
-  this._info = opts && opts.info;
+  this._defaultFullFlushFlag = fullFlush;
```

Comment:
This is fine, it mirrors the other flush flag fields (triple alliteration bonus!), I just wonder when it would be anything but `Z_FULL_FLUSH`? Is it different for brotli?

Reply:
Yes, brotli uses a different set of constants that don’t map 1:1 to zlib’s ones (`BROTLI_OPERATION_FLUSH` would be the one in this case)